python/tk_multi_publish2/dialog.py

```diff
@@ -230,23 +323,160 @@ def _update_details_from_selection(self):
-                self._create_task_details(publish_object)
-            elif isinstance(publish_object, Item):
+            if isinstance(publish_object, Item):
+                self._update_task_details_ui()
                 self._create_item_details(tree_item)
             elif publish_object is None:
+                self._update_task_details_ui()
                 # top node summary
                 self._create_master_summary_details()
 
+    def _update_task_details_ui(self, new_task_selection=None):
+        """
+        Updates the plugin UI widget.
+
+        This method should be called if everything is of the same type OR if the selection is
+        empty.
+
+        :param new_task_selection: A :class:`TaskSelection` containing the current UI selection.
+        """
+        new_task_selection = new_task_selection or self.TaskSelection()
+
+        # Nothing changed, so do nothing.
+        if self._current_tasks == new_task_selection:
+            return
+
+        # We're changing task, so we need to backup the current settings.
+        if self._current_tasks:
+            logger.debug("Saving settings...")
+            self._pull_settings_from_ui(self._current_tasks)
+
+        # If we're moving to a task that doesn't have a custom UI, clear everything.
+        if not new_task_selection:
+            # Note: At this point we don't really care if current task actually had a UI, we can
+            # certainly tear down an empty widget.
+            logger.debug("The ui is going to change, so clear the current one.")
+            self._clear_custom_settings_page()
+            self._current_tasks = new_task_selection
+            return
+
+        # A task was picked, so make sure our page is in foreground.
+        self.ui.details_stack.setCurrentIndex(self.TASK_DETAILS)
+
+        # If the new task selection does not have a custom UI, a simple tear down of the custom UI
+        # and setting the built-in fields will suffice.
+        if not new_task_selection.has_custom_ui:
+            logger.debug("Clearing custom UI and using default task details...")
+            self._clear_custom_settings_page()
+
+            # All the items are of the same type,
+            self.ui.settings_stack.setCurrentIndex(self.BUILTIN_TASK_DETAILS)
+            self.ui.task_icon.setPixmap(new_task_selection.plugin.icon)
+            self.ui.task_name.setText(new_task_selection.plugin.name)
+            self.ui.task_description.setText(new_task_selection.plugin.description)
+
+            # skip settings for now
+            # self.ui.task_settings.set_data(task.settings.values())
+
+            self._current_tasks = new_task_selection
+            return
+
+        # At this point we can assume we're going to have to show a UI, because new task exists
+        # and it has a custom UI.
+        self.ui.settings_stack.setCurrentIndex(self.CUSTOM_TASK_DETAILS)
+
+        # Now figure out if we need to create/replace the widgets.
+        if (
```

Comment:
this conditional is super confusing. suggest changing this to:
```python
if self.current_tasks and self._current_tasks.is_same_task_type(new_task_selection):
```
then simply flipping the logic in the conditional.

Reply:
This was addressed, but somehow Github didn't fold it. Oh well!